datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -100,6 +100,8 @@ fn is_false(expr: &Expr) -> bool {
 
 fn simplify(expr: &Expr) -> Expr {
     match expr {
+        Expr::BinaryExpr { left, .. } if is_literal_null(left) => *left.clone(),
```

Comment:
Thank you @Jimexist

I don't think this is true for `Operator::IsDistinctFrom` and `Operator::IsNotDistinctFrom` ,   `Operator::And` and `Operator::Or`,  so I think we should special case those Operators

Or perhaps to be more conservative make a whitelist of the arithmetic operators.


https://github.com/apache/arrow-datafusion/blob/81fae230b81b97e93bbb95b284cfda6f4d59552e/datafusion/src/logical_plan/operators.rs#L56-L58

examples

`null IS DISTINCT FROM null` --> `false` (but this PR would rewrite to `null` I think)

`true OR null` --> `true` (but this PR would rewrite it to `null` I think)